src/main/java/com/joy/joyfillandroid/TextField.kt

```diff
@@ -0,0 +1,175 @@
+import android.os.Build
+import android.view.Gravity
+import android.widget.LinearLayout
+import androidx.appcompat.widget.AppCompatTextView
+
+class TextField(context: Context?) : AppCompatTextView(context!!) {
+
+    private var borderColor: Int = 0
+    private var cornerRadius: Int = 0
+    private var borderWidth: Int = 0
+    private var isBold: Boolean = false
+    private var isItalic: Boolean = false
+    private var fontName: String? = null
+
+    companion object {
+        private const val DEFAULT_TEXT_SIZE_SP = 6f
+        private const val DEFAULT_TEXT_COLOR = android.R.color.black
+    }
+
+    init {
+        // Read attributes from XML
+
+        text = "Text Field"
+        textSize = TextField.DEFAULT_TEXT_SIZE_SP
+        val linearLayout = LinearLayout(context)
+        val layoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.WRAP_CONTENT,
+            LinearLayout.LayoutParams.WRAP_CONTENT
+        )
+        linearLayout.orientation = LinearLayout.VERTICAL
+        linearLayout.gravity = Gravity.CENTER_VERTICAL
+        linearLayout.layoutParams = layoutParams
+        layoutParams.setMargins(10,10,10,10)
+
+        setTextColor(Color.BLACK)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            typeface = Typeface.create(null, 500, false)
+        }
+        val padding = resources.getDimensionPixelSize(R.dimen.custom_edittext_padding)
+        setPadding(16, 0, 0, 10)
+
+        // Set default text color
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            setTextColor(context!!.getColor(TextField.DEFAULT_TEXT_COLOR))
+        }
+
+
+        val typedArray = context!!.obtainStyledAttributes(R.styleable.CustomTextView)
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            borderColor = typedArray.getColor(
+                R.styleable.CustomTextView_borderColor,
+                context!!.getColor(android.R.color.transparent)
+            )
+        }
+
+        cornerRadius = typedArray.getDimensionPixelSize(
+            R.styleable.CustomTextView_cornerRadius,
+            0
+        )
+
+        borderWidth = typedArray.getDimensionPixelSize(
+            R.styleable.CustomTextView_borderWidth,
+            0
+        )
+
+        isBold = typedArray.getBoolean(R.styleable.CustomTextView_boldText, false)
+        isItalic = typedArray.getBoolean(R.styleable.CustomTextView_italicText, false) //
+
+
+        applyProperties()
+
+        typedArray.recycle()
+    }
+
+
+    override fun onDraw(canvas: Canvas?) {
+        super.onDraw(canvas)
+        val layoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.WRAP_CONTENT
+        )
+        layoutParams.setMargins(30,10,30,10)
+        setLayoutParams(layoutParams)
+        layoutParams.gravity = Gravity.CENTER_VERTICAL
+    }
+
+
+    //MARK:- set Properties
+    private fun applyProperties() {
```

Comment:
So I noticed you have this apply properties here, but in the shortText field for instance you do not use something like this. If you plan to add it later then that is fine otherwise I would try to build things in a way that is consistent field to field.

This will help once you go to connect this field to the backend to pull all these values in dynamically one day you can do that the same way per field cause all fields are setup the same way with the exception of their specific use cases.